src/tests/helpers/random_values.c

```diff
@@ -0,0 +1,130 @@
+#include "../s21_decimal_test.h"
+
+static void set_random_pair_sign(s21_decimal *dec, mpz_t *in_mpz_copy);
+static void debug_print_pair(s21_decimal *dec, mpz_t *big, bool exp_applied);
+static void get_random_binary_string(s21_decimal *dec, char *mpz_bin_str, int size);
+
+int get_random_pair(s21_decimal *in, mpz_t *in_mpz_copy, int size) {
+    char mpz_bin_str[300] = {'\0'};
+
+    get_random_binary_string(in, mpz_bin_str, size);
+
+    int exp = get_rand(0, 28);
+    set_exponent(in, exp);
+
+    bool res = mpz_set_str(*in_mpz_copy, mpz_bin_str, 2);
+
+    /* This is heavily tested! According to output of binary calculator, we indeed */
+    /* get equal numbers (if we reverse our decimal representation) */
+
+    // debug_print_pair(in, in_mpz_copy, false);
+    apply_exponent_to_mpz(in_mpz_copy, exp);
+    // debug_print_pair(in, in_mpz_copy, true);
+
+    set_random_pair_sign(in, in_mpz_copy);
+
+    return res;
+}
+
+static void get_random_binary_string(s21_decimal *in, char *mpz_bin_str, int size) {
+    for (int j = 0, m = 0; j < size; j++) {
+        for (int i = 0; i < 32; i++, m++) {
+            if (rand() % 2) {
+                ADD_BIT(in->bits[j], i);
+                mpz_bin_str[m] = '1';
+            } else {
+                ZERO_BIT(in->bits[j], i);
+                mpz_bin_str[m] = '0';
+            }
+        }
+    }
+
+    /* This reverse is needed to match the format of normal binary with subject */
```

Comment:
Might need to remove this and restructure all mpz's to big endian & convert it to whatever endian once we are ready to give results back.